Branch-SDK/src/io/branch/indexing/ContentDiscoverer.java

```diff
@@ -151,83 +169,185 @@ public void run() {
+                }
+                if (view != null && view instanceof ViewGroup) {
+                    ViewGroup listView = (ViewGroup) view;
+                    JSONArray itemViewChildIdArray = viewKeyJson.getJSONArray(listViewID);
+                    int itemViewIds[] = new int[itemViewChildIdArray.length()];
+                    for (int i = 0; i < itemViewChildIdArray.length(); i++) {
+                        itemViewIds[i] = activity.getResources().getIdentifier(itemViewChildIdArray.getString(i), "id", activity.getPackageName());
+                    }
+                    int firstVisibleItemIdx = listView instanceof AbsListView ? ((AbsListView) listView).getFirstVisiblePosition() : 0;
+                    for (int j = 0; j < listView.getChildCount(); j++) {
+                        if (listView.getChildAt(j) != null) {
+                            JSONObject itemObj = new JSONObject();
+                            listViewContentObj.put("" + (j + firstVisibleItemIdx), itemObj);
+                            for (int i = 0; i < itemViewIds.length; i++) {
+                                if (listView.getChildAt(j) != null) {
+                                    View itemViewChild = listView.getChildAt(j).findViewById(itemViewIds[i]);
+                                    if (itemViewChild instanceof TextView) {
+                                        itemObj.put(itemViewChildIdArray.getString(i), getTextViewValue(itemViewChild, isClearText));
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    boolean isScrollListeningEnabled = viewKeyJson.has(ENABLE_SCROLL_WATCH) && viewKeyJson.getBoolean(ENABLE_SCROLL_WATCH);
+                    if (isScrollListeningEnabled && !viewTreeObserverMap.containsKey(viewKeyString)) {
+                        listView.getViewTreeObserver().addOnScrollChangedListener(scrollChangedListener);
+                        viewTreeObserverMap.put(viewKeyString, new WeakReference<>(listView.getViewTreeObserver()));
+                    }
                 }
-                updateElementData(viewName, childView, isClearText, contentDataArray, contentKeysArray);
             }
+            
+        } catch (JSONException e) {
+            e.printStackTrace();
         }
     }
-
+    
+    private ViewTreeObserver.OnScrollChangedListener scrollChangedListener = new ViewTreeObserver.OnScrollChangedListener() {
+        @Override
+        public void onScrollChanged() {
+            handler_.removeCallbacks(readListRunnable);
+            if (maxDiscoveryRepeatCnt > discoveryRepeatCnt_) {
+                handler_.postDelayed(readListRunnable, SCROLL_SETTLE_TIME);
+            }
+        }
+    };
+    private Runnable readListRunnable = new Runnable() {
+        @Override
+        public void run() {
+            readContentRunnable.run();
+        }
+    };
+    
```

Comment:
a little pet peeve but can you make the new lines between functions consistent? one new line ideally